compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -317,6 +327,24 @@ execCompile inputFile outputFile opts = withProjectRoot' (ProjectOpts Nothing (P
         createDirectoryIfMissing True $ takeDirectory outputFile
         B.writeFile outputFile $ Archive.encodeArchive bs
 
+execLint :: FilePath -> Options -> Command
+execLint inputFile opts =
+  withProjectRoot' (ProjectOpts Nothing (ProjectCheck "" False)) $ \relativize ->
+  do
+    loggerH <- getLogger opts "lint"
+    inputFile <- toNormalizedFilePath <$> relativize inputFile
+    opts' <- mkOptions opts
+    defaultHlintDataDir <-locateRunfiles $ mainWorkspace </> "compiler/damlc/daml-ide-core"
+    -- If not `--with-hlint path` given on the command line, assume
+    -- the testing default.
+    let opts'' =
+          case optHlintUsage opts' of
+            HlintEnabled _ -> opts'
+            HlintDisabled -> opts'{optHlintUsage=HlintEnabled defaultHlintDataDir}
+    withDamlIdeState opts'' loggerH diagnosticsLogger $ \ide -> do
+        setFilesOfInterest ide (Set.singleton inputFile)
+        runAction ide $ getHlintIdeas inputFile
```

Comment:
We should exit with `exitFailure` if we get diagnostics. An easy solution is probably something like:
```
diags <- getDiagnostics ide
unless (null diags) exitFailure
```

Reply:
✔️